scripts/generator/graphql_generator/csharp/SDK/Android/AndroidNativeCheckout.cs.erb

```diff
@@ -0,0 +1,94 @@
+        }
+
+        public bool CanShowPaymentSetup() {
+            // TODO
+            return true;
+        }
+
+        public void ShowPaymentSetup() {
+            // TODO
+        }
+
+        public bool CanCheckout(PaymentSettings paymentSettings) {
+            // TODO: Invoke methods in the plugin to determine if we can perform Android Pay
+            return true;
+        }
+
+        public void Checkout(
+            string key, 
+            ShopMetadata shopMetadata,
+            CheckoutSuccessCallback success, 
+            CheckoutCancelCallback cancelled, 
+            CheckoutFailureCallback failure
+        ) {
+            // TODO: Store callbacks and extract items we need from the cart to pass to Android Pay.
+            OnSuccess = success;
+            OnCancelled = cancelled;
+            OnFailure = failure;
+
+            var checkout = CartState.CurrentCheckout;
+
+            // TODO: Replace with Shop name
+            var merchantName = shopMetadata.Name;
+            var summaryItems = GetSummaryItemsFromCheckout(checkout);
+            var summaryString = Json.Serialize(summaryItems);
+            var currencyCodeString = checkout.currencyCode().ToString("G");
+            var countryCodeString = shopMetadata.PaymentSettings.countryCode().ToString("G");
+            var requiresShipping = checkout.requiresShipping();
+
+            #if !SHOPIFY_MONO_UNIT_TEST
+            using (var androidPayCheckoutSession = new AndroidJavaObject("com.shopify.unity.buy.AndroidPayCheckoutSession")) {
+                // TODO: Replace with mono object for receiving callbacks with.
+                object[] args = { "testing", merchantName, summaryString, currencyCodeString, countryCodeString, requiresShipping };
+                if (androidPayCheckoutSession.Call<bool>("checkoutWithAndroidPay", args)) {
+                    // TODO: Do I need to do anything on the Unity side after we've created the session
+                } else {
+                    // TODO: Create more meaningful error?
+                    var error = new ShopifyError(ShopifyError.ErrorType.NativePaymentProcessingError, "Unable to create Android Pay session.");
+                    OnFailure(error);
+                }
+            }
+            #endif
+        }
+
+        private List<SummaryItem> GetSummaryItemsFromCheckout(Checkout checkout) {
+            var summaryItems = new List<SummaryItem>();
+            summaryItems.Add(new SummaryItem("SUBTOTAL", checkout.subtotalPrice().ToString()));
+
+            if (checkout.requiresShipping()) {
+                try {
+                    summaryItems.Add(new SummaryItem("SHIPPING", checkout.shippingLine().price().ToString()));
+                } catch {}
+            }
+
+            summaryItems.Add(new SummaryItem("TAXES", checkout.totalTax().ToString()));
+            summaryItems.Add(new SummaryItem("TOTAL", checkout.totalPrice().ToString()));
+
+            return summaryItems;
```

Comment:
It was a List on iOS because Apple Pay takes an ordered list of summary items, but in this case Android Pay takes in individual values without the need for ordering